arch/x86/crc32_fold_pclmulqdq.c

```diff
@@ -274,6 +279,16 @@ Z_INTERNAL void crc32_fold_copy_pclmulqdq(crc32_fold *crc, uint8_t *dst, const u
         xmm_crc_part = _mm_setzero_si128();
     }
 
+#ifdef X86_VPCLMULQDQ_CRC
+    if (x86_cpu_has_vpclmulqdq && (len >= 256)) {
+        size_t n = fold_16_vpclmulqdq(&xmm_crc0, &xmm_crc1, &xmm_crc2, &xmm_crc3, dst, src, len);
+
+        len -= n;
+        src += n;
+        dst += n;
+    }
+#endif
+
     while (len >= 64) {
         crc32_fold_load((__m128i *)src, &xmm_t0, &xmm_t1, &xmm_t2, &xmm_t3);
 
```

Comment:
The lines below this one (evidently GH won't let me leave a comment far from your modifications) are a missed opportunity for a single 512 bit wide xor.  A lot of this code can have its loads and stores quartered.

Reply:
I don't know enough to be able to make those changes.

Reply:
I think I see what you mean with the xors. It might be a good idea to have an AVX512 version of this function in the future.